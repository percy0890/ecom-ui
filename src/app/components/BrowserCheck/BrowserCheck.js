import React, { Component } from 'react';
import Bowser from 'bowser';
import config from 'app/config/index.config';
import Modal from '../Modal';

class BrowserCheck extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isModalOpen: false,
    };

    this.minVersions = config.MIN_BROWSER_VERSIONS;
    const browser = Bowser.getParser(window.navigator.userAgent);

    if (!browser.satisfies(this.minVersions)) {
      this.state.isModalOpen = true;
    }
  }

  render() {
    return (
      <Modal
        isModalOpen={this.state.isModalOpen}
        closeModal={() => this.setState({ isModalOpen: false })}
      >
        <h2>Browser unsupported</h2>
      </Modal>
    );
  }
}

export default BrowserCheck;
