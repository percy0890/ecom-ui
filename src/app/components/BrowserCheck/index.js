import BrowserCheck from './BrowserCheck';
export default BrowserCheck;
