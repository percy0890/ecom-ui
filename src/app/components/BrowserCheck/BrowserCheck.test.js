import React from 'react';
import { shallow } from 'enzyme';
import BrowserCheck from '.';

describe('BrowserCheck Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<BrowserCheck />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should close Browser check warning modal on clicking Close', () => {
    wrapper.find('Modal').prop('closeModal')();
    expect(wrapper.state().isModalOpen).toEqual(false);
  });
});
