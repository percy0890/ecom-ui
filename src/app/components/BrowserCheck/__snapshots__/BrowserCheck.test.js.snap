// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserCheck Component renders correctly 1`] = `
<Modal
  closeModal={[Function]}
  isModalOpen={true}
>
  <h2>
    Browser unsupported
  </h2>
</Modal>
`;
